crates/amaru/src/bin/amaru/main.rs

```diff
@@ -29,28 +29,34 @@ enum Command {
 struct Cli {
     #[command(subcommand)]
     command: Command,
+    #[clap(long, short, action)]
+    disable_telemetry_export: bool,
 }
 
 #[tokio::main]
 async fn main() -> miette::Result<()> {
     panic_handler();
 
-    let (tracing, metrics, counter) = setup_tracing();
-
     let args = Cli::parse();
 
+    let (tracing, metrics, counter) = setup_tracing(args.disable_telemetry_export);
+
     let result = match args.command {
         Command::Daemon(args) => cmd::daemon::run(args, counter, metrics.clone()).await,
         Command::Import(args) => cmd::import::run(args).await,
     };
 
     // TODO: we might also want to integrate this into a graceful shutdown system, and into a panic hook
-    teardown_tracing(tracing, metrics)?;
+    if let Err(report) = teardown_tracing(tracing, metrics) {
+        eprintln!("Failed to teardown tracing: {report}");
+    }
 
     result
 }
 
-pub fn setup_tracing() -> (TracerProvider, SdkMeterProvider, Counter<u64>) {
+pub fn setup_tracing(
+    disable_telemetry_export: bool,
```

Comment:
I am not a huge fan of `bool` arguments in general, they are fragile, opaque, hard to interpret at call or use site, prone to errors...
Using a trait and 2 different implementations would be cleaner, eg.

```
trait LoggingSystem {
  fn setup();
  fn teardown();
}

struct SendToOTLP {
}

struct SendToStdout {
}

...
```
you get the idea I think :)

Reply:
Makes sense, although I would suggest a simpler struct container for now. It looks to me the whole telemetry/logging setup should be refactored more globally.

Reply:
Agreed. I trust your judgment, as long as we mark this as temporary and subject to further refactoring. IMO this whole OTLP thing should be completely externalised, with Amaru logging JSON to stdout by default and the environment doing "the right thing" to feed those into whatever o11y infrastructure is there.